muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java

```diff
@@ -135,19 +135,13 @@ public void saveBoolProperty(User user, String propertyName, boolean value) {
     userSchoolDataController.setUserProperty(user, "hops." + propertyName, value ? "yes" : "no");
   }
 
-  public boolean shouldShowStudies(UserEntity userEntity) {
-    if (userEntity != null) {
-      String studyViewStudents = pluginSettingsController.getPluginSetting("transcriptofrecords", "studyViewStudents");
-      if (studyViewStudents != null) {
-        String[] ids = studyViewStudents.split(",");
-        for (int i = 0; i < ids.length; i++) {
-          if (StringUtils.equals(ids[i], userEntity.getId().toString())) {
-            return true;
-          }
-        }
-      }
+  public boolean shouldShowStudies(User user) {
+    UserProperty userProperty = userSchoolDataController.getUserProperty(user, "hops.enabled");
```

Comment:
Apparently this method is used to determine if a view is shown in Muikku, so asking that based on a user variable in Pyramus is slow and rather illogical.

If any user-related information only affects Muikku, that information should be stored in Muikku (e.g. as a UserEntityProperty) as it's faster to retrieve and doesn't clutter Pyramus with data that has no relevance to Pyramus.

Reply:
Is there a place though in muikku to manage any such properties? I don't see this as major performance problem as (atm) it's used only in ToR and Pyramus has the facilities to manage the property by a user. In the end it's a property that should eventually be removed altogether.

Reply:
Fiiiiiine... Technically this can probably be left as-is, provided that the variable will eventually become obsolete. At least it follows the same behavior as all other HOPS related variables.

A separate Pyramus issue is probably needed to create the UserVariableKey itself, so that this information can then be managed over there. Irrelevant for this PR, though.